api/service/syncing/syncing.go

```diff
@@ -125,6 +125,21 @@ func (sc *SyncConfig) ForEachPeer(f func(peer *SyncPeerConfig) (brk bool)) {
 	}
 }
 
+// RemovePeer removes a peer from SyncConfig
+func (sc *SyncConfig) RemovePeer(peer *SyncPeerConfig) {
+	sc.mtx.Lock()
+	defer sc.mtx.Unlock()
+
+	peer.client.Close()
+	for i, p := range sc.peers {
+		if p == peer {
+			sc.peers = append(sc.peers[:i], sc.peers[i+1:]...)
```

Comment:
will this work without problem in a for loop? https://vbauerster.github.io/2017/04/removing-items-from-a-slice-while-iterating-in-go/

Reply:
Thanks for bring up this topic. You are right, removing multiple entries from a slice is not safe in the method. But the real use case here is that only one peer is removed at a time. The code itself does not do a good job in explaining itself. So following is my changes:

1. Break loop if the peer to remove is found.
2. Add duplicity check logic in both `AddPeer` and `SyncConfig.AddPeer` to ensure there is no duplicate in peers.

After the change, please review again.